_integration/testsuite/httpproxy/013-auth-basic-testserver.yaml

```diff
@@ -0,0 +1,311 @@
+
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: testserver
+  namespace: testserver-auth
+  labels:
+    app.kubernetes.io/name: testserver
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: testserver
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: testserver
+    spec:
+      containers:
+      - name: testserver
+        image: docker.io/projectcontour/contour-authserver:v2
+        imagePullPolicy: IfNotPresent
+        command:
+        - /contour-authserver
+        args:
+        - testserver
+        - --address=:9443
+        - --tls-ca-path=/tls/ca.crt
+        - --tls-cert-path=/tls/tls.crt
+        - --tls-key-path=/tls/tls.key
+        ports:
+        - name: auth
+          containerPort: 9443
+          protocol: TCP
+        volumeMounts:
+        - name: tls
+          mountPath: /tls
+          readOnly: true
+      volumes:
+      - name: tls
+        secret:
+          secretName: testserver
+
+---
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: testserver
+  namespace: testserver-auth
+  labels:
+    app.kubernetes.io/name: testserver
+spec:
+  ports:
+  - name: auth
+    protocol: TCP
+    port: 9443
+    targetPort: 9443
+  selector:
+    app.kubernetes.io/name: testserver
+  type: ClusterIP
+
+---
+
+apiVersion: projectcontour.io/v1alpha1
+kind: ExtensionService
+metadata:
+  name: testserver
+  namespace: testserver-auth
+spec:
+  services:
+  - name: testserver
+    port: 9443
+
+---
+
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: echo
+spec:
+  virtualhost:
+    fqdn: &fqdn echo.projectcontour.io
+    tls:
+      secretName: echo
+    authorization:
+      timeout: 500ms
+      extensionRef:
+        name: testserver
+        namespace: testserver-auth
+      authPolicy:
+        context:
+          hostname: *fqdn
+  routes:
+  - conditions:
+    - prefix: /first
+    authPolicy:
+      context:
+        target: first
+    services:
+    - name: echo
+      port: 80
+  - conditions:
+    - prefix: /second
+    authPolicy:
+      disabled: true
+    services:
+    - name: echo
+      port: 80
+  - services:
+    - name: echo
+      port: 80
+    authPolicy:
+      context:
+        target: default
+
+---
+
+import data.contour.http.client
+import data.contour.http.client.url
+import data.contour.http.response
+
+Response := client.Get({
+  "url": url.https(sprintf("/second/route/%d", [time.now_ns()])),
+  "headers": {
+    "Host": "echo.projectcontour.io",
+    "User-Agent": client.ua("auth-basic"),
+  },
+  "tls_insecure_skip_verify": true,
+})
+
+error_non_200_response [msg] {
+  not response.status_is(Response, 200)
+  msg := sprintf("got status %d, wanted %d", [Response.status_code, 200])
+}
+
+---
+
+import data.contour.http.client
+import data.contour.http.client.url
+import data.contour.http.response
+
+Response := client.Get({
+  "url": url.https(sprintf("/first/route/%d", [time.now_ns()])),
```

Comment:
it'd be helpful to add a comment somewhere explaining the authserver behavior that's being used here (authorize if path contains `allow`, otherwise deny)

Reply:
Good point, I added a comment.